lrauv_ignition_plugins/src/ScienceSensorsSystem.cc

```diff
@@ -634,22 +744,23 @@ void ScienceSensorsSystem::PostUpdate(const ignition::gazebo::UpdateInfo &_info,
       return;
     }
   }
+  else{
+    std::lock_guard<std::mutex> lock(this->dataPtr->dataMutex);
+    if (this->dataPtr->newDataAvailable)
```

Comment:
@arjo129 consider making `newDataAvailable` an `std::atomic_bool` to avoid locking a mutex on every simulation step. This flag will evaluate to `false` most of the time.

Reply:
To the best of my knowledge that would lead to incorrect behavior. Suppose multiple requests come in on the subscription topic, if this were purely an atomic there is no guarantee that the filepath will not have changed mid way through our read process. We may end up with a scenario where the latest file is not loaded.